### PR TITLE
[MGDAPI-3029] test: automate B01B - verify product logins

### PR DESCRIPTION
**Pull request by KevFan**
# Issue link

https://issues.redhat.com/browse/MGDAPI-3029
# What

Automating B01B test case for checking product logins for RHOAM for dedicated admin and developer users

# Verification steps

* Install RHOAM
```
export INSTALLATION_TYPE=managed-api
make cluster/deploy
```
* Once instllation completes, run the idp script
```
 PASSWORD=Password1 DEDICATED_ADMIN_PASSWORD=Password1 ./scripts/setup-sso-idp.sh
```
* Run the test
```
INSTALLATION_TYPE=managed-api BYPASS_STORAGE_TYPE_CHECK=true TEST="B01B" make test/e2e/single 
```
* Verify test passes

### CODE REVIEW

**Comment by psturc** [comment]
@KevFan  The test failed for the first time I ran it (fresh cluster, I haven't logged in with any other user than kube:admin to this cluster prior running the test).
```
[It] B01B - Verify users can login to products
  /Users/psturc/work/integr8ly/integreatly-operator/test/functional/integreatly_test.go:77
not creating IDP, it's already created
Oauth Proxy to https://grafana-route-redhat-rhoam-customer-monitoring-operator.<REDACTED>.org for user test-user01

• Failure [16.929 seconds]
integreatly
/Users/psturc/work/integr8ly/integreatly-operator/test/functional/integreatly_test.go:11
  managed-api IDP BASED
  /Users/psturc/work/integr8ly/integreatly-operator/test/functional/integreatly_test.go:74
    B01B - Verify users can login to products [It]
    /Users/psturc/work/integr8ly/integreatly-operator/test/functional/integreatly_test.go:77

    Expected 403 error for user test-user01 but didn't: <nil>
```

Following 3 test runs were successful. I'm going to recreate the SSO IDP on the cluster to see if I can reproduce the test failure.

**Comment by psturc** [comment]
@KevFan  I was able to reproduce the error twice in a row - this error always happens if the test is run right after SSO IDP is set up.

(before reconfiguring SSO IDP, I'm using these commands to do a cleanup after SSO IDP is removed from OCM)
```bash
oc delete user --all
oc delete identity --all
oc delete keycloakuser --all -n redhat-rhoam-user-sso
oc delete keycloakuser --all -n redhat-rhoam-rhsso
oc delete keycloakclient testing-idp-client -n redhat-rhoam-rhsso
oc delete keycloakrealm testing-idp -n redhat-rhoam-rhsso
```

**Comment by KevFan** [comment]
@psturc Thanks Pavel, good spot :+1: I'll have a look to see what might be the issue :thinking:

**Comment by KevFan** [comment]
@psturc Issue for why `test-user01` didn't see 403 on first login was due to it needs to accept the permissions form and this returns `nil` whether it's redirected to 403 page or not. Subsequent logins will not need to accept pernissions and will give an `err` with the 403 in the `ProxyOAuth`  function
* https://github.com/integr8ly/integreatly-operator/blob/077981a7bea3dffad9109714e22bef0a17e1fa78/test/resources/oauth_proxy_auth.go#L115-L231
I've refactored the test code to use chromeDP actions and check for this on initial login so hopefully it should be fixed now if you want to have a look :crossed_fingers:

**Comment by psturc** [comment]
taking a look again

File: test-cases/tests/authorization/b04b-verify-dedicated-admin-user-permissions-are-correct.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
```
Comment (psturc) [inline test-cases/tests/authorization/b04b-verify-dedicated-admin-user-permissions-are-correct.md:1]: Should [this test case](https://github.com/integr8ly/integreatly-operator/blob/5ccc7d3fa5954d2f673cb8f0f90ff0f9d3f206bf/test-cases/tests/authorization/b01b-verify-that-the-users-can-login-in-all-products.md) be updated instead?

Comment (KevFan) [reply test-cases/tests/authorization/b04b-verify-dedicated-admin-user-permissions-are-correct.md:1]: Good catch, completely updated the wrong test case :sweat_smile:

**Comment by KevFan** [comment]
@psturc I think the Dockerfile change should allow the test run on the built docker image now. `quay.io/kevfan/integreatly-operator-test-harness:latest` is the image build anyway that I've built that just runs this single test if you want to try it out 
[integreatly-operator-test--1-52qdb-test.log](https://github.com/integr8ly/integreatly-operator/files/7726906/integreatly-operator-test--1-52qdb-test.log)

Do you think this would also affect the pipelines or would it just run this built docker image ? :thinking:

**Comment by psturc** [comment]
> @psturc I think the Dockerfile change should allow the test run on the built docker image now. `quay.io/kevfan/integreatly-operator-test-harness:latest` is the image that I've built that just runs this single test if you want to try it out [integreatly-operator-test--1-52qdb-test.log](https://github.com/integr8ly/integreatly-operator/files/7726906/integreatly-operator-test--1-52qdb-test.log)
> 
> Do you think this would also affect the pipelines or would it just run this built docker image ? 🤔

@KevFan I ran the test the same way that pipelines do (using delorean CLI) and it worked. So I don't think this will affect our pipelines. Nice work!

/lgtm
/approve

**Pull request merged**